fairscale/nn/data_parallel/fully_sharded_data_parallel.py

```diff
@@ -1680,7 +1680,7 @@ def _post_backward_hook(self, param: Parameter, *unused: Any) -> None:
         # then subsequent hook callbacks will see POST state.
         self.assert_state([TrainingState.BACKWARD_PRE, TrainingState.BACKWARD_POST])
         self.training_state = TrainingState.BACKWARD_POST
-        if param.grad is None:
+        if param.grad is None and param.main_grad is None:
```

Comment:
Just checking if we want to have

`if getattr(param, "main_grad", None) is None` instead of `param.main_grad is None` to be safer